awx/plugins/inventory/foreman.py

```diff
@@ -203,6 +259,49 @@ def _get_json(self, url, ignore_errors=None, params=None):
                 break
         return results
 
+    def _use_inventory_report(self):
+         url = "%s/api/v2/status" % self.foreman_url
+         result = self._get_json(url)
+         foreman_version = LooseVersion(result.get('version')) >= LooseVersion('1.24.0')
+         return foreman_version and self.foreman_use_reports_api
+
+    def _fetch_params(self):
+        options, params = ("no", "yes") , dict()
+        params["Organization"] = options[self.want_organization]
+        params["Location"] = options[self.want_location]
+        params["IPv4"] = options[self.want_IPv4]
+        params["IPv6"] = options[self.want_IPv6]
+        params["Facts"] = options[self.want_facts]
+        params["Host Group"] = options[self.want_host_group]
+        params["Host Collections"] = options[self.want_hostcollections]
+        params["Subnet"] = options[self.want_subnet]
+        params["Subnet v6"] = options[self.want_subnet_v6]
+        params["Smart Proxies"] = options[self.want_smart_proxies]
+        params["Content Attributes"] = options[self.want_content_facet_attributes]
+        params["Hosts"] = options[self.host_filters]
```

Comment:
@sjha4 This line throws "TypeError: tuple indices must be integers, not NoneType" when "host_filters" not used in ini file (and "TypeError: tuple indices must be integers, not str" when used).

Reply:
This should work now.